Sources/ReactiveArray.swift

```diff
@@ -3,20 +3,25 @@ import ReactiveSwift
 import Result
 
 public final class ReactiveArray<Element> {
-
 	public typealias Snapshot = ContiguousArray<Element>
 	public typealias Change = Delta<Snapshot, IndexSet>
 
-	public let signal: Signal<Change, NoError>
+	fileprivate let storage: Storage<ContiguousArray<Element>>
```

Comment:
> Can't we just use Atomic as our storage? Reads may not need to be protected but it's not harmful either.

Every unnecessary locking burns joules and CPU time. In this specific case, reads can be lock free, and arrays are expected to have superior read performance. So why not let it be?

Moreover, unless you gonna reimplement all Sequence and Collection algorithms, they all rely on the subscript. If you fuel the subscript with `Atomic`, that means even the simplest iterator or `map` gonna lock once every element.